src/connector/main.sol

```diff
@@ -0,0 +1,439 @@
+    function handleOperation(
+        address token,
+        uint256 amount,
+        uint256 fee,
+        LeverageMacroOperation memory operation
+    ) external {
+        require(msg.sender == address(flashLoanReceiver));
+
+        uint256 beforeSwapsLength = operation.swapsBefore.length;
+        if (beforeSwapsLength > 0) {
+            _doSwaps(operation.swapsBefore);
+        }
+
+        // Based on the type we do stuff
+        if (operation.operationType == OperationType.OpenCdpOperation) {
+            _openCdpCallback(operation.OperationData);
+        }/* else if (operation.operationType == OperationType.OpenCdpForOperation) {
+            _openCdpForCallback(operation.OperationData);
+        } else if (operation.operationType == OperationType.CloseCdpOperation) {
+            _closeCdpCallback(operation.OperationData);
+        } else if (operation.operationType == OperationType.AdjustCdpOperation) {
+            _adjustCdpCallback(operation.OperationData);
+        } else if (operation.operationType == OperationType.ClaimSurplusOperation) {
+            _claimSurplusCallback();
+        }*/
+
+        uint256 afterSwapsLength = operation.swapsAfter.length;
+        if (afterSwapsLength > 0) {
+            _doSwaps(operation.swapsAfter);
+        }
+
+        IERC20(token).transfer(address(flashLoanReceiver), amount + fee);
+    }
+
+    /// @dev Must be memory since we had to decode it
+    function _doSwaps(SwapOperation[] memory swapData) internal {
+        uint256 swapLength = swapData.length;
+
+        for (uint256 i; i < swapLength; ) {
+            _doSwap(swapData[i]);
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// @dev Given a SwapOperation
+    ///     Approves the `addressForApprove` for the exact amount
+    ///     Calls `addressForSwap`
+    ///     Resets the approval of `addressForApprove`
+    ///     Performs validation via `_doSwapChecks`
+    function _doSwap(SwapOperation memory swapData) internal {
+        // Ensure call is safe
+        // Block all system contracts
+        //_ensureNotSystem(swapData.addressForSwap);
+
+        // Exact approve
+        // Approve can be given anywhere because this is a router, and after call we will delete all approvals
+        IERC20(swapData.tokenForSwap).safeApprove(
+            swapData.addressForApprove,
+            swapData.exactApproveAmount
+        );
+
+        // Call and perform swap
+        // NOTE: Technically approval may be different from target, something to keep in mind
+        // Call target are limited
+        // But technically you could approve w/e you want here, this is fine because the contract is a router and will not hold user funds
+        (bool success, ) = excessivelySafeCall(
+            swapData.addressForSwap,
+            gasleft(),
+            0,
+            0,
+            swapData.calldataForSwap
+        );
+        require(success, "Call has failed");
```

Comment:
Arbitrary call, are sure you want to keep this in?
This increases the surface area by a lot